apigw-lambda-healthscribe-sam/README.md

```diff
@@ -0,0 +1,173 @@
+   Adding the new layer to your Lambda function's configuration. Please wait ...
+   {
+      "FunctionName": "{ClinicalNotesGenerationLambdaFunction}",
+      ......
+      ......
+      "State": "Active",
+      "LastUpdateStatus": "InProgress",
+      "LastUpdateStatusReason": "The function is being created.",
+      "LastUpdateStatusReasonCode": "Creating",
+      "PackageType": "Zip",
+      "Architectures": [
+         "arm64"
+      ],
+      "EphemeralStorage": {
+         "Size": 512
+      }
+   }      
+   ```    
+10. You may have to press `q` to come out of the output. The setup is ready for testing.
+
+## How it works
+
+
+Please refer to the architecture diagram below:
+
+![End to End Architecture](images/architecture.png)
+
+Here's a breakdown of the steps:
+
+1. **Amazon API Gateway**: Receives the HTTP POST request. API Gateway forwards the request to the AWS Lambda function.
+
+2. **AWS Lambda**: The function calls AWS HealthScribe API using boto3 to start the medical scribe job. 
+
+3. **AWS HealthScribe**: Analyses the patient-clinician conversations audio file from Amazon S3 bucket and generates clinical notes from patient-clinician conversations. It then uploads the the clinical note in a JSON format into a given S3 bucket.
+
+
+## Testing
+
+1. Upload the `conversation.mp3` file into the Amazon S3 bucket using the following command. Please replace `TranscribeNotesBucket` with the value from the `sam deploy --guided` output:
+
+   ```bash
+   aws s3 cp conversation.mp3 s3://{TranscribeNotesBucket}/conversation.mp3
+   ```
+
+2. We will use [curl](https://curl.se/) to send a HTTP POST request to the API. Make sure to replace `GenerateClinicalNotesAPI` with the one from your `sam deploy --guided` output:
+
+   ```bash
+   curl -X POST {GenerateClinicalNotesAPI}
+   ```
+
+   The API returns AWS HealthScribe `jobName`. Please make a note of it, we will use it on the next step. 
+
+   ```json
+   {"jobStatus": "IN_PROGRESS", "jobName": "3dcxxxx-xxxx-xxxx-xxxx-xxxx0fff25"}
+   ```
+
+3. Execute the below command to get the current job status. Please replace the `jobName` from the previous step and `your-region` with your deployment region. Please wait till the status is `COMPLETED` before moving to next step. 
+   ```bash
+   aws transcribe get-medical-scribe-job  --medical-scribe-job-name {jobName} --query 'MedicalScribeJob.MedicalScribeJobStatus' --region {your-region}
+   ```
+
+4. Execute the below command to download the generated clinical notes by AWS HealthScribe from the Amazon S3 bucket. Please replace `TranscribeNotesBucket` with the value from the `sam deploy --guided` output and `jobName` from the previous step. Please validate the downloaded `summary.json` content. Please refer to the `SummarizedSegment` tags under different segments.
+   ```bash
+   aws s3 cp s3://{TranscribeNotesBucket}/{jobName}/summary.json .
+   ```
+
+
+## Cleanup
+
+1. Delete content of the Amazon S3 bucket using the following command:
```

Comment:
```suggestion
1. Delete the content in the Amazon S3 bucket using the following command:
```

Reply:
Add note to double check the bucket name as this is a destructive operation

Reply:
Added, updated.